riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/HttpClientFactory.java

```diff
@@ -97,8 +97,8 @@ public static CloseableHttpClient createHttpClient(final Client client,
         return builder.build();
     }
 
-    private static CachingHttpClientBuilder configureCaching(final Caching caching,
-            @Nullable final HttpCacheStorage cacheStorage) {
+    private static HttpClientBuilder configureCaching(final Caching caching,
+            @Nullable final Object cacheStorage) {
```

Comment:
Can be already typed here and the cast can be pushed upwards to the public method.

Reply:
Not really. If I leave HttpCacheStorage here I'm receiving:
```
Failed to introspect Class [org.zalando.riptide.autoconfigure.HttpClientFactory] from ClassLoader [jdk.internal.loader.ClassLoaders$AppClassLoader@2c13da15]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:507) ~[spring-core-5.1.8.RELEASE.jar:5.1.8.RELEASE]
```

Seems that even if the method is private somehow spring still tries to evaluate it.